cmd/versioninfo/versioninfo.go

```diff
@@ -0,0 +1,39 @@
+package versioninfo
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	// GitBranch is the branch from which this binary was built
+	GitBranch string
+	// ReleaseTag is the git tag from which this binary is released
+	ReleaseTag string
+	// BuildTime is the time at which this binary was built
+	BuildTime string
+	// GitTreeState indicates if the git tree, from which this binary was built, was clean or dirty
+	GitTreeState string
+	// GitCommit is the git commit at which this binary binary was built
+	GitCommit string
+	// GitMajor is the major version of the release
+	GitMajor string
+	// GitMinor is the minor version of the release
+	GitMinor string
+)
+
+// VersionCmd is the version command for the binary
+func VersionCmd() *cobra.Command { // nolint
+	return &cobra.Command{
+		Use:   "version",
+		Short: "Print version of this binary",
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Printf("Git Branch: %q\n", GitBranch)
+			fmt.Printf("Git commit: %q\n", GitCommit)
+			fmt.Printf("Release tag: %q\n", ReleaseTag)
+			fmt.Printf("Git tree state: %q\n", GitTreeState)
+			fmt.Printf("Git version: Major:%q, Minor:%q\n", GitMajor, GitMinor)
```

Comment:
This output appears misleading for alpha releases (such as the current one).

I'd like to see the output changed up to:

- Have a simplified output by default
  - If git tree state is clean and release tag commit is the current commit, print the tagged version
  - If git tree state is dirty or the release tag commit is not the current commit, output a modified simple version
- An optional verbose output that gave more details about the version, such as the current output

Reply:
Here's what the version output looks like:
```
$ clusterawsadm version
Version Info: GitReleaseTag: "v1.0.0-alpha.3", MajorVersion: "1", MinorVersion:"0", GitReleaseCommit:"1d5a033234f40b", GitTreeState:"dirty"
$ clusterawsadm version -l
Version Info:
GitReleaseTag: "v1.0.0-alpha.3", Major: "1", Minor: "0", GitRelaseCommit: "1d5a033234f40b"
Git Branch: "version-cmd"
Git commit: "b18ba239425057-dirty"
Git tree state: "dirty"
$ clusterawsadm version --long
Version Info:
GitReleaseTag: "v1.0.0-alpha.3", Major: "1", Minor: "0", GitRelaseCommit: "1d5a033234f40b"
Git Branch: "version-cmd"
Git commit: "b18ba239425057-dirty"
Git tree state: "dirty"
$ clusterawsadm version foo
Error: accepts 0 arg(s), received 1
Usage:
  clusterawsadm version [flags]

Flags:
  -h, --help   help for version
  -l, --long   Print longhand version info

accepts 0 arg(s), received 1
```